query-request.hh

```diff
@@ -9,8 +9,12 @@
 
 #pragma once
 
+#include <memory>
 #include <optional>
 
+#include "cql3/functions/function.hh"
+#include "cql3/functions/function_name.hh"
+#include "cql3/functions/aggregate_function.hh"
```

Comment:
It's not good for query-request to reach into cql/ and pull things from there.

We need to create a new abstraction for functions, and have both cql/ and here depend on it.

Reply:
I've moved this to db/, however only minimal required subset of cql's function (`function`, `function_name`, `aggregate_function`). Also there are aliases to old cql types.